cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -369,8 +369,20 @@ func ValidateHostPort(endpoint string, fldPath *field.Path) field.ErrorList {
 	return allErrs
 }
 
+func validateMaximumServiceSubnetSize(cidr *net.IPNet, maxCIDRBits int) error {
+	// Should be a small cidr due to how it is stored in etcd.
+	// bigger cidr (specially those offered by IPv6) will add no value
+	// significantly increase snapshotting time.
+	// NOTE: This is identical to validation performed in the apiserver.
+	var ones, bits = cidr.Mask.Size()
+	if bits-ones > maxCIDRBits {
+		return errors.Errorf("specified service subnet is too large; for %d-bit addresses, the mask must be >= %d", bits, bits-maxCIDRBits)
+	}
+	return nil
+}
+
 // ValidateIPNetFromString validates network portion of ip address
-func ValidateIPNetFromString(subnetStr string, minAddrs int64, isDualStack bool, fldPath *field.Path) field.ErrorList {
+func ValidateIPNetFromString(subnetStr string, minAddrs int64, isDualStack, isServiceSubnet bool, fldPath *field.Path) field.ErrorList {
```

Comment:
I will not overload current function and add a new one like `ValidateServiceSubnetSize()`
this will allow to unit test it and will be easier to add new code like in 
https://github.com/kubernetes/kubernetes/pull/95723/files#diff-71c615dc50838bd457062893d9f12906d130be2ea2f1ce964317dc43ca82ad2bR407